### ISSUE TITLE
Document with a table name not able to utilize

### ISSUE THREAD

**Issue by harshalizode** [dymmond/mongoz#40]
Hello,
I have tried to rename my collection as per the documentation [https://mongoz.tarsild.io/documents/?h=collection#document-without-table-name](url), but here is an issue.
When I am trying to add the collection in the Meta class then, it is raising the issue as
```
    class Company(BaseDocument):
  File "/home/venv/proj.env/lib/python3.11/site-packages/mongoz/core/db/documents/metaclasses.py", line 290, in __new__
    collection_name = meta.collection.name
                      ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'name'
```

**Comment by tarsil** [comment]
@harshalizode hello l. Can you provide a proper example please? To be able to replicate

**Comment by harshalizode** [comment]
Yes,
I have the model as
```

class Company(BaseDocument):
    """
        **Summary** This Class represents the company.
            **Fields:**
                - code (str): It contains the company code.
                - name (str): It contains the company name.
                - branches (List[CompanyBranch]): It contains the embedded \
                        documents of company_branch model.
    """
    code: str = mongoz.String
    display_name: str = mongoz.String(null=True)
    branches: List[CompanyBranch] = mongoz.Array(CompanyBranch, null=True)

    objects: ClassVar[CompanyQueryManager] = CompanyQueryManager()

    def __str__(self):
        return f"{self.display_name} ({self.code})"

    class Meta:
        collection = "companies"
        indexes = [
            mongoz.Index("code", unique=True),
        ]
```
Instead of appending the's' suffix to the model name, I have overridden the collection name in accordance with the documentation of Mongoz.

With this class wherever Itry to start my application it raises the `AttributeError` as indicated above..

**Comment by tarsil** [comment]
I have indeed an error (regression) but not there. Actually, in your example you are also missing `code: str = mongoz.String()` the `()`.

**Comment by tarsil** [comment]
@harshalizode can you provide me the full example, please? You have a base document + embedded as well

**Comment by tarsil** [comment]
@harshalizode I was able to replicate 100%. It was a regression but it was supposed to be a release anyway. That is now addressed in the release 0.10.0 👍🏼

**Comment by harshalizode** [comment]
Yes, It is resolved in the latest version :)
